google_symptoms/delphi_google_symptoms/date_utils.py

```diff
@@ -0,0 +1,130 @@
+"""utility functions for date parsing."""
+
+from datetime import date, datetime, timedelta
+from itertools import product
+from typing import Dict, List, Tuple
+
+import covidcast
+from delphi_utils.validator.utils import lag_converter
+from pandas import to_datetime
+
+from .constants import COMBINED_METRIC, FULL_BKFILL_START_DATE, PAD_DAYS, SMOOTHERS
+
+
+def generate_patch_dates(params) -> List[Dict[date, List[date]]]:
+    """
+    Generate date range for chunking backfilled dates.
+
+    Parameters
+    ----------
+    params
+
+    Returns
+    -------
+    list of dicts
+    """
+    issue_date = datetime.strptime(params["patch"]["start_issue"], "%Y-%m-%d")
+
+    # If end_date not specified, use current date.
+    end_date = datetime.strptime(params["patch"]["end_issue"], "%Y-%m-%d")
+
+    patch_dates_list = []
+
+    max_expected_lag = lag_converter(params["validation"]["common"].get("max_expected_lag", {"all": 4}))
+    global_max_expected_lag = max(list(max_expected_lag.values()))
+    num_export_days = params["validation"]["common"].get("span_length", 14) + global_max_expected_lag
+
+    while issue_date <= end_date:
+        # start_date gets padded again when executed in run_module
+        expected_start_dt = issue_date - timedelta(days=num_export_days - PAD_DAYS + 2)
+        daterange = generate_date_range(expected_start_dt, issue_date, num_export_days)
+        patch_date = {issue_date: daterange}
+        patch_dates_list.append(patch_date)
+        issue_date += timedelta(days=1)
+    return patch_dates_list
+
+
+def _generate_candidate_dates(params: Dict, logger) -> Tuple[date, date, int]:
+    """
+    Generate candidate export_start_date and candidate_export_end_date with possible lag.
```

Comment:
I was trying to align the date parsing/ export date generating steps consistent of how it was produced which was to generate a candidate start date, end date, and number of export dates 
and then call generate_date_range

since it's not the final dates that's used in the bigquery I named it candidates, since that was the most nearby concepts I could think.